pycytominer/cyto_utils/load.py

```diff
@@ -1,8 +1,66 @@
 import csv
 import gzip
+from pathlib import Path
 import numpy as np
 import pandas as pd
 
+def infer_profile_file_type(file):
+    """Infers profile file type.
+
+    This is done by reading the header of the given file to identify
+    the file type.
+
+    Currently this function only identifies sqlite and parquet
+    headers.
+
+    Parameters
+    ----------
+    file : str
+        file name
+
+    Return
+    -------
+    str
```

Comment:
The care you provided through type hints within the docstring here made me wonder if it might be helpful to include within the function definition as well. Doing this can be a matter of ["gradual typing"](https://peps.python.org/pep-0483/#summary-of-gradual-typing).